From b437c65815b2898564748f3be48c4145671a750f Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Tue, 2 Jul 2024 14:48:09 +0200
Subject: [PATCH] backends/drm: disable triple buffering on NVidia

Until we figure out what's wrong, it's best to leave it disabled

BUG: 487833
---
 src/backends/drm/drm_output.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/drm/drm_output.cpp b/src/backends/drm/drm_output.cpp
index 68d43d7ae42..7561c899168 100644
--- a/src/backends/drm/drm_output.cpp
+++ b/src/backends/drm/drm_output.cpp
@@ -45,7 +45,7 @@ DrmOutput::DrmOutput(const std::shared_ptr<DrmConnector> &conn)
 {
     m_pipeline->setOutput(this);
     m_renderLoop->setRefreshRate(m_pipeline->mode()->refreshRate());
-    if (m_gpu->atomicModeSetting() && !s_disableTripleBuffering) {
+    if (m_gpu->atomicModeSetting() && !s_disableTripleBuffering && !m_gpu->isNVidia()) {
         m_renderLoop->setMaxPendingFrameCount(2);
     }
 
-- 
GitLab

